gson/src/main/java/com/google/gson/DateFormatType.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gson;
+
+public enum DateFormatType {
+	/**
+	 * Use custom {@link java.text.DateFormat} to format Date at serialization if
+	 * was set. Otherwise is same as {@code DateFormatType.DEFAULT}
+	 */
+	CUSTOM,
```

Comment:
Not sure if we need this level of customization. We could instead just provide a few sample DateTypeAdapters that match up with these that users can register.
Why add more complexity to the API?